salt/cache/mysql_cache.py

```diff
@@ -118,7 +118,8 @@ def run_query(conn, query, retries=3, args=None):
         else:
             log.info("mysql_cache: recreating db connection due to: %r", e)
         __context__["mysql_client"] = MySQLdb.connect(**__context__["mysql_kwargs"])
-        return run_query(__context__["mysql_client"], query, args, retries - 1)
+        return run_query(conn=__context__.get("mysql_client"),
+                query=query, args=args, retries=(retries - 1))
```

Comment:
Just nitpicking at this point, but since the `conn` and `query` args are positional, we might want to pass them in as such.

```suggestion
        return run_query(__context__.get("mysql_client"),
                query, args=args, retries=(retries - 1))
```

Reply:
Makes sense.